### ISSUE TITLE
[3.4] Hyphens in slugs throw exceptions

### ISSUE THREAD

**Issue by bobdenotter** [bolt/bolt#7124]
If you have a contenttype like this:

```yml
dummies:
    slug: dummies-foo
    name: Dummies
    singular_slug: dummy-foo
    singular_name: Dummy
    fields:
        title:
            type: text
            class: large
            group: content
        slug:
            type: slug
            uses: title
        teaser:
            type: html
            height: 150px
```

It throws an exception:

![screen shot 2017-10-26 at 15 19 56](https://user-images.githubusercontent.com/1833361/32055150-3b0f45ca-ba61-11e7-9398-f551151e5d02.png)

Curiously enough, this _does_ work: 

```yml
dummies:
    slug: dummiesfoo
    name: Dummies
    singular_slug: dummyfoo
    singular_name: Dummy
    fields:
        title:
            type: text
            class: large
            group: content
        slug:
            type: slug
            uses: title
        teaser:
            type: html
            height: 150px
```

**Comment by bobdenotter** [comment]
pinging @rossriley

**Comment by bobdenotter** [comment]
Fixed in #7140
